libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -162,4 +166,42 @@ export class JobRepository extends BaseRepository<EnforceEnvironmentQuery, JobEn
       .lean()
       .exec();
   }
+
+  public async shouldDelayDigestJobOrMerge(job: JobEntity): Promise<{ matched: number; modified: number }> {
+    const execution = {
+      matched: 0,
+      modified: 0,
+    };
+
+    const delayedDigestJobs = await this._model.find({
+      status: JobStatusEnum.DELAYED,
+      type: StepTypeEnum.DIGEST,
+      _templateId: job._templateId,
+      _environmentId: this.convertStringToObjectId(job._environmentId),
+      _subscriberId: this.convertStringToObjectId(job._subscriberId),
+    });
+
+    const matched = delayedDigestJobs.length;
+    execution.matched = matched;
+
+    if (execution.matched === 0) {
+      const updatedDigestJob = await this._model.updateOne(
+        {
+          _environmentId: job._environmentId,
+          _templateId: job._templateId,
+          _subscriberId: job._subscriberId,
+          _id: job._id,
+        },
+        {
+          $set: {
+            status: JobStatusEnum.DELAYED,
+          },
+        }
+      );
+
+      execution.modified = updatedDigestJob.modifiedCount;
+    }
+
+    return execution;
+  }
```

Comment:
We moved the previous domain logic to the DAL layer to be able to fully lock it at DAL layer with the distributed lock. Unfortunately I haven't been able to implement with Mongoose/MongoDB functionalities the logic: 
find if there is a existing delayed job for this subscriber-template-enviroment -> if not found make this certain job as the digest one delayed.
That was my goal. But weren't able to fix it with workflow changes or MongoDB functionalities (API, transactions and locks, etc)

That logic was implemented before in the domain. When concurrent calls hit it, there was a race condition as both calls when trying to find a existing delayed job didn't find anything, creating more than one single delayed job, therefore digest not working properly and duplicating the number of notifications sent.